src/main/java/at/hannibal2/skyhanni/features/mining/ActualGemstonePowderDisplay.kt

```diff
@@ -0,0 +1,285 @@
+            val basePercent = atomizedCrystals * 100
+            val echoPercent = echoOfAtomizedRate * 100
+
+            val totalAddedPercent = basePercent + (basePercent * echoOfAtomizedRate)
+
+            val sb = StringBuilder()
+            sb.append("§7 Attributes: §a+${totalAddedPercent.formatPretty()}%")
+
+            if (echoOfAtomizedRate > 0.0) {
+                val baseStr = basePercent.formatPretty()
+                val echoStr = echoPercent.formatPretty()
+                sb.append(" §7($baseStr% + $echoStr% of $baseStr%)")
+            }
+
+            lines.add(sb.toString())
+        }
+
+        if (is2xEventActive) {
+            lines.add("§7 2x Powder Event: §a+100%")
+        }
+
+        if (isSkymallActive) {
+            lines.add("§7 Sky Mall: §a+15%")
+        }
+
+        if (hotmBuffPercent > 0) {
+            lines.add("§7 HOTM Powder Buff: §a+${hotmBuffPercent}%")
+        }
+
+        val drillBonusPercent = (drillMultiplier - 1.0) * 100.0
+        if (drillBonusPercent > 0.0) {
+            lines.add("§7 Drill: §a+${drillBonusPercent.formatPretty()}%")
+        }
+
+        val petBonusPercent = (petMultiplier - 1.0) * 100.0
+        if (petBonusPercent > 0.0 && petName != null) {
+            lines.add("§7 $petName Pet: §a+${petBonusPercent.formatPretty()}%")
+        }
+
+        lines.add("")
+        lines.add("§7Total Multiplier: §6${totalMultiplier.formatPretty()}x")
+        lines.add("§7Effective Powder: §d${effectiveAmount.addSeparators()}")
+
+        return lines
+    }
+
+    @HandleEvent(priority = HandleEvent.LOW)
+    fun onChat(event: SkyHanniChatEvent) {
+        if (!isEnabled()) return
+
+        PowderChestReward.GEMSTONE_POWDER.chatPattern.matchMatcher(event.message) {
+            val amountStr = groupOrNull("amount") ?: return
+            val originalAmount = amountStr.formatInt()
+
+            val breakdown = calculateMultiplierBreakdown()
+
+            val effectiveAmount = (originalAmount * breakdown.totalMultiplier).toInt()
+
+            ChatUtils.debug("[PowderDebug] Base: $originalAmount, Effective: $effectiveAmount, Multiplier: ${breakdown.totalMultiplier}")
+
+            if (breakdown.totalMultiplier > 1.0) {
+                val originalFormatted = originalAmount.addSeparators()
+                val effectiveFormatted = effectiveAmount.addSeparators()
+
+                val hoverText = breakdown.toTooltipLines(originalAmount, effectiveAmount)
+
+                event.chatComponent = TextHelper.text("    §r§dGemstone Powder §r§8x$originalFormatted §7(x$effectiveFormatted)") {
+                    this.hover = TextHelper.multiline(hoverText)
+                }
+            }
+        }
+    }
+
+
+    private fun getDrillMultiplierFraction(): Double {
+        var bonusPercent = 0
+        val heldItem = InventoryUtils.getItemInHand() ?: return 1.0
+
+        val internalName = heldItem.getInternalName()
+        if (!internalName.contains("DRILL")) return 1.0
```

Comment:
This check does the same, but does not need to compare Strings, therefore it is a bit faster in general.
```suggestion
        if(heldItem.getItemCategoryOrNull() != ItemCategory.DRILL)
```

Reply:
👍